ScarletEngine/Sources/ECS/Public/Registry.h

```diff
@@ -200,47 +86,107 @@ namespace ScarletEngine
 		void SortAll()
 		{
 			ZoneScoped
-			for (auto& Container : ComponentContainers)
+			for (const auto& [_, Container] : ComponentContainers)
 			{
-				Container.second->Sort();
+				Container->Sort();
 			}
 		}
 
-		template <typename T>
-		T* GetSingleton() const
+		template <typename ComponentType>
+		ComponentType* GetSingleton() const
 		{
-			static T Instance;
+			// HACK: Sort of a hacky way to implement singleton components but very performant since it doesn't use any hashmaps
+			static ComponentType Instance;
 			return &Instance;
 		}
 
-		const Array<SharedPtr<Entity>>& GetEntities() const { return Entities; }
+		void Clear()
+		{
+			// Note: this does NOT clear any values for singletons.
+			ComponentContainers.clear();
+			Entities.clear();
+			NextAvailableEID = 1;
+		}
+
+		template <typename ...ComponentTypes>
+        Array<ProxyType<ComponentTypes...>> GetProxies() const
```

Comment:
why is their a weird indentation? is that only github preview?

Reply:
Its github preview because one line is using spaces and the other line is using tabs. I will fix this before submit.